frontend/src/lib/components/MapComponent.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+	import { onMount, onDestroy, setContext, createEventDispatcher, tick } from 'svelte';
+	import L from 'leaflet';
+	import 'leaflet/dist/leaflet.css';
+
+	export let bounds: L.LatLngBoundsExpression | undefined = undefined;
+	export let view: L.LatLngExpression | undefined = undefined;
+	export let zoom: number | undefined = undefined;
+
+	const dispatch = createEventDispatcher();
+
+	let map: L.Map | undefined;
+	let mapElement: HTMLElement;
+
+	onMount(() => {
+		if (!bounds && (!view || !zoom)) {
+			throw new Error('Must set either bounds, or view and zoom.');
+		}
+
+		map = L.map(mapElement)
+			.on('zoom', (e) => dispatch('zoom', e))
+			.on('popupopen', async (e) => {
```

Comment:
You can remove the 'zoom' and 'popupopen' listeners for now.  We can add them back in if the need arises.